src/components/astro/GoogleDriveExportButton.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Actions } from 'jumpstate';
+import Button from 'grommet/components/Button';
+import { config } from '../../lib/config';
+
+const STATUS = {
+  IDLE: 'idle',
+  CONFIGURING: 'configuring',
+  SUCCESS: 'success',
+  ERROR: 'error',
+};
+
+let GoogleAuth = null;
+
+class GoogleDriveExportButton extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      isAuthorized: false,
+      status: STATUS.IDLE
+    }
+
+    this.handleError = this.handleError.bind(this);
+    this.setupGoogleClient = this.setupGoogleClient.bind(this);
+    this.tryExport = this.tryExport.bind(this);
+    this.updateSigninStatus = this.updateSigninStatus.bind(this);
+  }
+
+  componentDidMount() {
+    this.setupGoogleClient();
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (nextState.isAuthorized !== this.state.isAuthorized) return false;
+
+    return true;
+  }
+
+  componentWillUnmount() {
+    if (this.state.isAuthorized) GoogleAuth.signOut();
+    const script = document.querySelector('#google-api');
+    script.parentNode.removeChild(script);
+  }
+
+  updateSigninStatus(isSignedIn) {
+    this.setState({ isAuthorized: isSignedIn }, this.sendAuthorizedRequest);
+  }
+
+  handleError(error) {
+    console.error(error);
+    this.setState({ error: STATUS.ERROR })
+  }
+
+  setupGoogleClient() {
+    if (!document.querySelector('#google-api')) {
+      this.setState({ status: STATUS.CONFIGURING });
+      const script = document.createElement('script');
+      script.src = config.google;
+      script.id = 'google-api';
+
+      script.onload = () => {
+        const gapi = window.gapi;
```

Comment:
I take it that it's safe to assume that when the `<script>` loads, we can be guaranteed that `window.gapi` is there. At the moment I can't think of any scenario where the script loads but .gapi is removed or invalid, so that's 👌 , but let me know if you can spot otherwise.

Reply:
Yeah, I think the only scenario that I can think of is if the gapi client is deprecated and if it is, then it will probably fail to load anyway.